pkg/v1/providers/infrastructure-vsphere/v1.1.0/ytt/overlay.yaml

```diff
@@ -1,14 +1,60 @@
 #@ load("@ytt:overlay", "overlay")
 #@ load("@ytt:data", "data")
 #@ load("@ytt:regexp", "regexp")
+#@ load("@ytt:assert", "assert")
 
 
+#@ valid_pci_deviceid = ["1EB8"]
+
+#@ valid_pci_vendorid = ["10DE"]
+
+#@ strateg_type = ["OnDelete", "RollingUpdate"]
+
 #@ load("lib/helpers.star", "get_bom_data_for_tkr_name", "get_default_tkg_bom_data", "kubeadm_image_repo", "get_image_repo_for_component", "get_vsphere_thumbprint", "get_tkr_version_from_tkr_name")
 
 #@ load("lib/validate.star", "validate_configuration")
 #@ load("@ytt:yaml", "yaml")
 #@ validate_configuration("vsphere")
 
+#@ def verify_and_configure_strategy(strategytype):
+#@   if strategytype not in strateg_type:
+#@     assert.fail("Invalid Strategy type  \""+strategytype+"\", it must be OnDelete or RollingUpdate")
+#@   end
+#@   return strategytype
+#@ end
+
+
+
+#@ def get_pci_devices(pciDevicesString, pciIgnorepciDeviceValidation):
+#@   for val in pciDevicesString.split(','):
+#@     if len(val) <= 0:
+#@        assert.fail("given labels string \""+pciDevicesString+"\" must be in the  \"vendorID:deviceID,vendorID:deviceID,..\" format ")
+#@      end
+#@     kv = val.split(':')
+#@     if len(kv) != 2:
+#@        assert.fail("given labels string \""+pciDevicesString+"\" must be in the  \"vendorID:deviceID,vendorID:deviceID,..\" format ")
+#@      end
+#@     if not pciIgnorepciDeviceValidation and (kv[0] not in valid_pci_vendorid or kv[1] not in valid_pci_deviceid):
```

Comment:
This looks good only when a device Id is universally unique. If they are not unique, this validation may go wrong. For example, let's say we have vendors V1 & V2; V1 has device with Ids D1, D2 & V2 has device with Ids D2 & D3. If we are going to support only V1's D2 & V2's D3, this logic will allow V2's D2 as well.